konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/impl/step/ml/classifier/ClassifierOutputRunner.java

```diff
@@ -0,0 +1,201 @@
+ *  * License for the specific language governing permissions and limitations
+ *  * under the License.
+ *  *
+ *  * SPDX-License-Identifier: Apache-2.0
+ *  *****************************************************************************
+ */
+
+package ai.konduit.serving.pipeline.impl.step.ml.classifier;
+
+import ai.konduit.serving.annotation.runner.CanRun;
+import ai.konduit.serving.pipeline.api.context.Context;
+import ai.konduit.serving.pipeline.api.data.Data;
+import ai.konduit.serving.pipeline.api.data.NDArray;
+import ai.konduit.serving.pipeline.api.data.ValueType;
+import ai.konduit.serving.pipeline.api.step.PipelineStep;
+import ai.konduit.serving.pipeline.api.step.PipelineStepRunner;
+import ai.konduit.serving.pipeline.util.DataUtils;
+import lombok.AllArgsConstructor;
+
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.DoubleStream;
+
+@AllArgsConstructor
+@CanRun(ClassifierOutputStep.class)
+public class ClassifierOutputRunner implements PipelineStepRunner {
+
+    protected final ClassifierOutputStep step;
+
+
+    @Override
+    public void close() {
+
+    }
+
+    @Override
+    public PipelineStep getPipelineStep() {
+        return step;
+    }
+
+    @Override
+    public Data exec(Context ctx, Data data) {
+
+        String inputName = step.inputName();
+
+
+        if (inputName == null) {
+            String errMultipleKeys = "NDArray field name was not provided and could not be inferred: multiple NDArray fields exist: %s and %s";
+            String errNoKeys = "NDArray field name was not provided and could not be inferred: no image NDArray exist";
+            inputName = DataUtils.inferField(data, ValueType.NDARRAY, false, errMultipleKeys, errNoKeys);
+        }
+
+
+        NDArray classifierOutput = data.getNDArray(inputName);
+
+        boolean batch = false;
+        if (classifierOutput.shape().length == 2 && classifierOutput.shape()[0] > 1) {
+            batch = true;
+        }
+
+        // If not specified, the predicted class index as a string is used - i.e., "0", "1"
+        List<String> labels = step.Labels();
+
+        if (labels == null) {
+            labels = new ArrayList<String>();
+        }
+        if (labels.isEmpty()) {
+            for (int i = 0; i < classifierOutput.shape()[1]; i++) {
+                labels.add(Integer.toString(i));
+            }
+        }
+
+
+        if (!batch) {
+            double[] classifierOutputArr = squeze(classifierOutput);
+            double[] maxValueWithIdx = getMaxValueAndIndex(classifierOutputArr);
+            double prob = maxValueWithIdx[0];
+            long index = (long) maxValueWithIdx[1];
+            String label = labels.get((int) index);
+
+            if (step.topN() != null && step.topN() > 1) {
+                if (step.returnProb()) {
+                    data.putListDouble(step.probName(), Collections.singletonList(prob));
```

Comment:
Technically these prob/index/labelName field can be null, in sp
Most of the time they won't, but they could be under the following circumstances:
1.  `new ClassifierOutputStep().probName(null);`
2. Manually written JSON
3. JSON created by one of the other clients (python, javascript, etc)

so: Let's switch to something like this for all of the names:
```
String probName = step.probName() == null ? ClassifierOutputStep.DEFAULT_PROB_NAME : step.probName()
```

And let's do that once earlier (to avoid code duplication). Or a simple method that's used in each place would be fine too.